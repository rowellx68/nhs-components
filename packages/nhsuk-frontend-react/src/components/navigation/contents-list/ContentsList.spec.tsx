import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, { Default as DefaultStory } from './ContentsList.stories';

const Default = composeStory(DefaultStory, meta);

it('should render a navigation element with a contents list', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('.nhsuk-contents-list')).toBeInTheDocument();
  expect(container.querySelectorAll('a')).toHaveLength(4);
  expect(container).toMatchSnapshot();
});
