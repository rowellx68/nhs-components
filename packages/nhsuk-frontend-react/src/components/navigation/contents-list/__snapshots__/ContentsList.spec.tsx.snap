// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a navigation element with a contents list 1`] = `
<div>
  <nav
    aria-label="Pages in this guide"
    class="nhsuk-contents-list"
    role="navigation"
  >
    <h2
      class="nhsuk-u-visually-hidden"
    >
      Contents
    </h2>
    <ol
      class="nhsuk-contents-list__list"
    >
      <li
        aria-current="page"
        class="nhsuk-contents-list__item"
      >
        <span
          class="nhsuk-contents-list__current"
          href="#"
        >
          What is AMD?
        </span>
      </li>
      <li
        class="nhsuk-contents-list__item"
      >
        <a
          class="nhsuk-contents-list__link"
          href="#"
        >
          Symptoms
        </a>
      </li>
      <li
        class="nhsuk-contents-list__item"
      >
        <a
          class="nhsuk-contents-list__link"
          href="#"
        >
          Getting diagnosed
        </a>
      </li>
      <li
        class="nhsuk-contents-list__item"
      >
        <a
          class="nhsuk-contents-list__link"
          href="#"
        >
          Treatment
        </a>
      </li>
      <li
        class="nhsuk-contents-list__item"
      >
        <a
          class="nhsuk-contents-list__link"
          href="#"
        >
          Living with AMD
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
